components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/prompt/PromptInstanceUpdateDelegate.kt

```diff
@@ -0,0 +1,36 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.browser.engine.gecko.prompt
+
+import mozilla.components.browser.engine.gecko.GeckoEngineSession
+import mozilla.components.concept.engine.prompt.PromptRequest
+import org.mozilla.geckoview.GeckoSession
+
+/**
+ * Implementation of [GeckoSession.PromptDelegate.PromptInstanceDelegate] used to update a
+ * prompt request when onPromptUpdate is invoked.
+ *
+ * @param geckoSession [GeckoEngineSession] used to notify the engine observer
+ * with the onPromptUpdate callback.
+ * @param previousPromptRequestUID UID of the previous prompt request.
+ * @param updatePrompt Method used to obtain a [PromptRequest]
+ * from the received [GeckoSession.PromptDelegate.BasePrompt].
+ */
+internal class PromptInstanceUpdateDelegate(
```

Comment:
This only applicable for `ChoicePrompt` I think we should adapt the name to reflect that.